libbpf-rs/src/map.rs

```diff
@@ -59,6 +59,35 @@ impl OpenMap {
     pub fn set_inner_map_fd(&mut self, inner: &Map) {
         unsafe { libbpf_sys::bpf_map__set_inner_map_fd(self.ptr, inner.fd()) };
     }
+
+    /// Reuse an already-pinned map for `self`.
+    pub fn reuse_pinned_map<P: AsRef<Path>>(&mut self, path: P) -> Result<()> {
+        let cstring = util::path_to_cstring(path)?;
+
+        let fd = unsafe { libbpf_sys::bpf_obj_get(cstring.as_ptr()) };
+        if fd < 0 {
+            return Err(Error::System(errno::errno()));
+        }
+
+        let ret = unsafe { libbpf_sys::bpf_map__reuse_fd(self.ptr, fd) };
+
+        // Always close `fd` regardless of if `bpf_map__reuse_fd` succeeded or failed
+        match unistd::close(fd) {
+            Ok(_) => (),
+            Err(e) => {
+                return Err(Error::Internal(format!(
+                    "Failed to close pinned map fd: {}",
+                    e
+                )))
+            }
+        }
```

Comment:
there isn't much you can do if close() fails, and in practice it shouldn't ever fail, so I'd drop this error handling part